pkg/providers/aws/cluster_vpc.go

```diff
@@ -460,11 +453,22 @@ func getSubnets(ec2Svc ec2iface.EC2API) ([]*ec2.Subnet, error) {
 }
 
 // function to get vpc of a cluster
-func getVpc(ctx context.Context, c client.Client, ec2Svc ec2iface.EC2API, logger *logrus.Entry) (*ec2.Vpc, error) {
-	// get vpcs
-	vpcs, err := ec2Svc.DescribeVpcs(&ec2.DescribeVpcsInput{})
+func getClusterVpc(ctx context.Context, c client.Client, ec2Svc ec2iface.EC2API, logger *logrus.Entry) (*ec2.Vpc, error) {
+	// first call to aws api from the network provider is to get cluster vpc
+	// polling to allow credential minter time to reconcile credentials
+	var vpcs []*ec2.Vpc
+	var pollErr error
+	err := wait.PollImmediate(time.Second*5, time.Minute*5, func() (done bool, err error) {
+		vpcOutput, err := ec2Svc.DescribeVpcs(&ec2.DescribeVpcsInput{})
+		if err != nil {
+			pollErr = err
+			return false, nil
+		}
+		vpcs = vpcOutput.Vpcs
+		return true, nil
+	})
```

Comment:
The credential minter should only need one chance to reconcile new credentials, right? Is it worth having a poll here when we can just error out and try again in the next reconcile loop?

Reply:
yeah very true, just an oversight, this change has been reverted